ros2_batch_job/__main__.py

```diff
@@ -441,7 +447,52 @@ def run(args, build_function, blacklisted_package_names=None):
         # Print the pip version
         job.run(['"%s"' % job.python, '-m', 'pip', '--version'], shell=True)
         # Install pip dependencies
-        job.run(['"%s"' % job.python, '-m', 'pip', 'install', '-U'] + pip_dependencies, shell=True)
+        install_pip_packages = list(pip_dependencies)
+        if not args.colcon_branch:
+            install_pip_packages += colcon_packages
+        job.run(
+            ['"%s"' % job.python, '-m', 'pip', 'install', '-U'] +
+            install_pip_packages, shell=True)
+
+        # OS X can't invoke a file which has a space in the shebang line
+        # therefore invoking vcs explicitly through Python
+        if args.do_venv:
+            vcs_cmd = [
+                '"%s"' % job.python,
+                '"%s"' % os.path.join(venv_path, 'bin', 'vcs')]
+        else:
+            vcs_cmd = ['vcs']
+
+        if args.colcon_branch:
+            # create .repos file for colcon repositories
+            os.makedirs('colcon', exist_ok=True)
+            with open('colcon/colcon.repos', 'w') as h:
```

Comment:
Could we just use existing repos file rather than constructing it ? https://raw.githubusercontent.com/colcon/colcon.readthedocs.org/master/colcon.repos

Is this because of the packages that are platform specific ?

Reply:
> Could we just use existing repos file rather than constructing it ? 

No, because the set is not identical.

> Is this because of the packages that are platform specific ?

The platform specific ones are another reason why the file is being generated.

Reply:
sounds good :+1: